SmartPointers/shared_pointer.hpp

```diff
@@ -1,80 +1,134 @@
-//
-//  shared_pointer.hpp
-//  Pointer
-//
-//  Created by Srushti Desai on 12/27/23.
-//
+#include <utility>
+#include <iostream>
+#include <stdio.h>
 
-#ifndef shared_pointer_hpp
-#define shared_pointer_hpp
+template <typename T> class SharedPointer {
 
-using namespace std;
-#include <iostream>
-template<class T>
-class SharedPointer
-{
-private:
-    T* m_ptr;
-    unsigned int* m_count;
-public:
-     SharedPointer():m_ptr(nullptr), m_count(new unsigned int(0))
-    {
-        
+  public:
+    // Constructor
+    SharedPointer() : m_ptr(nullptr), count(new int(1)) {
+        std::cout << "Called Shared Constructor \n";
     }
 
-    ~SharedPointer(){
-        //cleanup();
-        
+    // Paramatized Constructor
+    SharedPointer(T* ptr) : m_ptr(ptr), count(new int(1)) {
+        std::cout << "called  Shared Paramatized Constructor \n";
     }
 
-    SharedPointer(T * ptr) : m_ptr(ptr), m_count(new unsigned int(1))
-        {
+    // Copy Constructor
```

Comment:
Please remove all of these comments. These comments are extra overhead on brain and don't add any extra information

Reply:
solved